test/server.js

```diff
@@ -13,9 +13,61 @@ const config = {
     port: 5432,
     name: 'sensorthingstest',
     user: 'postgres',
-    pass: '12345678'
+    pass: '12345678',
+    hooks: {
+      'beforeValidate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeValidate: 'beforeValidate'
+        });
+      },
+      'afterValidate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterValidate: 'afterValidate'
+        });
+      },
+      'beforeCreate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeCreate: 'beforeCreate'
+        });
+      },
+      'afterCreate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterCreate: 'afterCreate'
+        });
+      },
+      'beforeDestroy': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeDestroy: 'beforeDestroy'
+        });
+      },
+      'afterDestroy': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterDestroy: 'afterDestroy'
+        });
+      },
+      'beforeRestore': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeRestore: 'beforeRestore'
+        });
+      },
+      'afterRestore': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterRestore: 'afterRestore'
+        });
+      },
+      'beforeUpdate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeUpdate: 'beforeUpdate'
+        });
+      },
+      'afterUpdate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterUpdate: 'afterUpdate'
+        });
+      }
+    }
   }
```

Comment:
I wasn't able to initialize the db hooks only for the "db hooks" test (because it is a singleton, I don't see a way to initialize the db with hooks for the "db hooks" test and without db hooks for all the other tests). Therefore, many tests fail because instances have properties the tests are not expecting. I will think about possible solutions but I would appreciate your input on this issue.